timemachine/fe/free_energy.py

```diff
@@ -1461,6 +1463,7 @@ def get_flattened_params(initial_state: InitialState) -> NDArray:
     last_update_time = begin_loop_time
 
     for current_frame in range(md_params.n_frames):
+        per_state_water_sampling = [(0, 0) for _ in range(len(initial_states))]
```

Comment:
nit: could this name be more descriptive? I don't have great ideas; e.g. `water_sampling_acceptance_proposal_counts_by_state` seems a bit long but maybe still preferable?